libs/live-config/src/FeatureFlags/firebaseFeatureFlags.ts

```diff
@@ -0,0 +1,110 @@
+import { PropsWithChildren } from "react";
+import { snakeCase } from "lodash";
+import semver from "semver";
+import { Feature, FeatureId } from "@ledgerhq/types-live";
+import { getEnv } from "@ledgerhq/live-env";
+import { LiveConfig } from "./LiveConfig";
+
+export type FirebaseFeatureFlagsProviderProps = PropsWithChildren<unknown>;
+export const formatToFirebaseFeatureId = (id: string) => `feature_${snakeCase(id)}`;
+
+export const checkFeatureFlagVersion = (feature: Feature) => {
+  const platform = LiveConfig.getInstance().platform;
+  if (feature && feature.enabled && platform) {
+    let featureSpecificVersion: string | undefined;
+    if (platform === "desktop") {
+      featureSpecificVersion = feature.desktop_version;
+    }
+    if (["ios", "android"].includes(platform)) {
+      featureSpecificVersion = feature.mobile_version;
+    }
+    if (
+      featureSpecificVersion &&
+      !semver.satisfies(LiveConfig.getInstance().appVersion, featureSpecificVersion, {
+        includePrerelease: true,
+      })
+    ) {
+      return {
+        enabledOverriddenForCurrentVersion: true,
+        ...feature,
+        enabled: false,
+      };
+    }
+  }
+  return feature;
+};
```

Comment:
Improves readability and code quality through use of early return and switch / case instead of if cascade 

```suggestion
export const checkFeatureFlagVersion = (feature: Feature): Feature => {
  const platform = LiveConfig.getInstance().platform;

  if (!feature || !feature.enabled || !platform) {
    return feature;
  }

  const featureSpecificVersion: string | undefined = (() => {
    switch (platform) {
      case "desktop":
        return feature.desktop_version;
      case "ios":
      case "android":
        return feature.mobile_version;
      default:
        return undefined;
    }
  })();

  if (
    featureSpecificVersion &&
    !semver.satisfies(LiveConfig.getInstance().appVersion, featureSpecificVersion, {
      includePrerelease: true,
    })
  ) {
    return {
      enabledOverriddenForCurrentVersion: true,
      ...feature,
      enabled: false,
    };
  }

  return feature;
};
```

Reply:
fixed